functions/Get-DbaBackupHistory.ps1

```diff
@@ -1,449 +1,441 @@
+                        $wherearray += "database_name in ('$dblist')"
+                    }
+                    
+                    if ($Last -or $LastFull -or $LastLog -or $LastDiff)
+                    {
+                        $tempwhere = $wherearray -join " and "
+                        $wherearray += "type = 'Full' and mediaset.media_set_id = (select top 1 mediaset.media_set_id $from $tempwhere order by backupset.backup_finish_date DESC)"
+                    }
+                    
+                    if ($Since -ne $null)
+                    {
+                        $wherearray += "backupset.backup_finish_date >= '$since'"
+                    }
+                    
+                    if ($IgnoreCopyOnly)
+                    {
+                        $wherearray += "is_copy_only='0'"
+                    }
+                    
+                    if ($where.length -gt 0)
+                    {
+                        $wherearray = $wherearray -join " and "
+                        $where = "$where $wherearray"
+                    }
+                    
+                    $sql = "$select $from $where ORDER BY backupset.backup_finish_date DESC"
+                }
+                
+                if (!$last)
+                {
+                    Write-Message -Level Debug -Message $sql
+                    Write-Message -Level SomewhatVerbose -Message "Executing sql query"
+                    $results = $server.ConnectionContext.ExecuteWithResults($sql).Tables.Rows | Select-Object * -ExcludeProperty BackupSetRank, RowError, Rowstate, table, itemarray, haserrors
+                    if ($raw)
+                    {
+                        Write-Message -Level SomewhatVerbose -Message "Processing as Raw Ouput"
+                        $results = $results | Select-Object *, @{ Name = "FullName"; Expression = { $_.Path } }
+                        Write-Message -Level SomewhatVerbose -Message "$($results.Count) result sets found"
+                    }
+                    else
+                    {
+                        Write-Message -Level SomewhatVerbose -Message "Processing as Grouped output"
+                        $GroupedResults = $results | Group-Object -Property backupsetid
+                        Write-Message -Level SomewhatVerbose -Message "$($GroupedResults.Count) result-groups found"
+                        $GroupResults = @()
+                        foreach ($group in $GroupedResults)
+                        {
+                            
+                            $FileSql = @"
+select
+	file_type as FileType,
+	logical_name as LogicalName,
+	physical_name as PhysicalName
+from
+    msdb.dbo.backupfile
+where
+    backup_set_id='$($Group.group[0].BackupSetID)'
+"@
+                            Write-Message -Level Debug -Message "FileSQL: $FileSql"
+                            
+                            $obj = New-Object sqlcollective.dbatools.Database.BackupHistory
```

Comment:
Done, truly, a bad name in general.
Used to it because I myself always use it in a very restricted situation (Exactly as here: filling in properties on a result object before passing it to an array). Personal habit, so I didn't consider it on a project-wide scale.